Mention NaN handling in dtype description

jowagner opened the pull request:
To achieve preservation and avoid interpretation of string or object dtypes, NaN value interpretation must be switched off.

- [x] closes issue #20875
- [ ] tests passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] `pandas.read_csv.html` renders text as intended

jowagner committed:
Mention NaN handling in dtype description

To achieve preservation and avoid interpretation of string or object dtypes, NaN value interpretation must be switched off.

jreback commented on pandas/io/parsers.py:
```diff
@@ -125,7 +125,8 @@
 dtype : Type name or dict of column -> type, default None
     Data type for data or columns. E.g. {'a': np.float64, 'b': np.int32}
-    Use `str` or `object` to preserve and not interpret dtype.
+    Use `str` or `object` together with passing `keep_default_na=False` and
```
this is too prescriptive here, rather you can put a pointer to the NA Values section

jowagner replied:
Thanks. I made a new commit to my branch.

jowagner committed:
reduce detail in reference to NaN settings

Being less prescriptive how to use dtype=str as per suggestion from @jreback

gfyoung commented on pandas/io/parsers.py:
```diff
@@ -125,7 +125,8 @@
     Data type for data or columns. E.g. {'a': np.float64, 'b': np.int32}
-    Use `str` or `object` to preserve and not interpret dtype.
+    Use `str` or `object` together with suitable `na_values` settings
+    to preserve and not interpret dtype.
```
Can we add a newline below this?

gfyoung replied:
It's minor enough that I wouldn't consider that an issue.

jreback commented on pandas/io/parsers.py:
```diff
@@ -125,7 +125,8 @@
     Data type for data or columns. E.g. {'a': np.float64, 'b': np.int32}
-    Use `str` or `object` to preserve and not interpret dtype.
+    Use `str` or `object` together with suitable `na_values` settings
+    to preserve and not interpret dtype.
```
can you add the same test in io.rst as well

jowagner committed:
mention NaN handling in dtype description

Apply changes to the dtype description in io/parser.py (mentioning NaN handling) also to io.rst.

jowagner committed:
Fix rst usage error in io.rst

Not sure what role to use with a parameter reference. Using a literal for the moment.

TomAugspurger commented:
CircleCI failures is fixed on master.

Thanks @jowagner.

jowagner commented:
And thanks for your patience while I learned how to do this.
